components/ResourceBalance.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {useMemo} from "react";
+import {
+  GOLD_TOKEN_ADDRESS,
+  GRAIN_TOKEN_ADDRESS,
+  GRASS_TOKEN_ADDRESS,
+  IRON_TOKEN_ADDRESS, SILVER_TOKEN_ADDRESS,
+  WATER_TOKEN_ADDRESS, WOOD_TOKEN_ADDRESS, WOOL_TOKEN_ADDRESS
+} from "constants/addresses";
+import useSWR from "swr";
+import {BaseErc20Factory} from "@zoralabs/core/dist/typechain";
+import {defaultProvider} from "constants/providers";
+import {useWeb3React} from "@web3-react/core";
+import {formatUnits} from '@ethersproject/units';
+
+const resourceToContract: {
+  [key: string]: string
+} = {
+  water: WATER_TOKEN_ADDRESS,
+  gold: GOLD_TOKEN_ADDRESS,
+  grain: GRAIN_TOKEN_ADDRESS,
+  grass: GRASS_TOKEN_ADDRESS,
+  iron: IRON_TOKEN_ADDRESS,
+  silver: SILVER_TOKEN_ADDRESS,
+  wood: WOOD_TOKEN_ADDRESS,
+  wool: WOOL_TOKEN_ADDRESS,
+
+}
+
+export const ResourceBalance: React.FC<{ resource: string }> = ({resource}) => {
+
+  const {account} = useWeb3React()
+  const contractAddress = useMemo(() => resourceToContract?.[resource.toLowerCase()], [resource])
```

Comment:
`resourceToContract` should always be defined? why `?`

Reply:
ah yeah typescript complains if not optional because i haven't value typed the attributes array